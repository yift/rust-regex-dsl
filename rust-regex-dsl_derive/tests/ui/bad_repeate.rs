pub use rust_regex_dsl::regex_dsl;

fn main() {
    let regex = regex_dsl! {
        "|",
        maybe,
        "|",
    };
    println!("{}", regex);
    let regex = regex_dsl! {
        "|",
        maybe_repeat {
            12
        },
        "|",
    };
    println!("{}", regex);
    let regex = regex_dsl! {
        "|",
        repeat {
            lazy, lazy,
            "test"
        },
        "|",
    };
    println!("{}", regex);
    let regex = regex_dsl! {
        "|",
        repeat {
            lazy, greedy,
            "test"
        },
        "|",
    };
    println!("{}", regex);
    let regex = regex_dsl! {
        "|",
        repeat {
            greedy, greedy
            "test"
        },
        "|",
    };
    println!("{}", regex);
    let regex = regex_dsl! {
        "|",
        repeat {
            greedy, lazy
            "test"
        },
        "|",
    };
    println!("{}", regex);
}