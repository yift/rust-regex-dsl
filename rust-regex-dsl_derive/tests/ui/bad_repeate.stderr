error: Unknown word: maybe
 --> tests/ui/bad_repeate.rs:6:9
  |
6 |         maybe,
  |         ^^^^^

error: expected one of: string literal, character literal, identifier, `#`, `~`
  --> tests/ui/bad_repeate.rs:13:13
   |
13 |             12
   |             ^^

error: Can not set quantifier type twice
  --> tests/ui/bad_repeate.rs:21:23
   |
21 |             lazy, lazy,
   |                       ^

error: Can not set quantifier type twice
  --> tests/ui/bad_repeate.rs:30:25
   |
30 |             lazy, greedy,
   |                         ^

error: Can not set quantifier type twice
  --> tests/ui/bad_repeate.rs:40:13
   |
40 |             "test"
   |             ^^^^^^

error: Can not set quantifier type twice
  --> tests/ui/bad_repeate.rs:49:13
   |
49 |             "test"
   |             ^^^^^^
